WeakAuras/Transmission.lua

```diff
@@ -316,39 +270,287 @@ ChatFrame_AddMessageEventFilter("CHAT_MSG_BN_WHISPER_INFORM", filterFunc)
 ChatFrame_AddMessageEventFilter("CHAT_MSG_INSTANCE_CHAT_LEADER", filterFunc)
 
-function ShowTooltip(content)
-  if(ItemRefTooltip.WeakAuras_Tooltip_Thumbnail) then
-    ItemRefTooltip.WeakAuras_Tooltip_Thumbnail:Hide();
-    ItemRefTooltip.WeakAuras_Tooltip_Thumbnail = nil;
-  end
-  if(ItemRefTooltip.WeakAuras_Tooltip_Thumbnail_Frame) then
-    ItemRefTooltip.WeakAuras_Tooltip_Thumbnail_Frame:Hide();
+local buttonAnchor = CreateFrame("FRAME", "WeakAurasTooltipAnchor", ItemRefTooltip)
+buttonAnchor:SetAllPoints()
+buttonAnchor:Hide()
+buttonAnchor:RegisterEvent("PLAYER_REGEN_ENABLED")
+buttonAnchor:RegisterEvent("PLAYER_REGEN_DISABLED")
+
+local thumbnailAnchor = CreateFrame("FRAME",  "WeakAurasTooltipThumbnailFrame", buttonAnchor)
+thumbnailAnchor:SetSize(40,40)
+thumbnailAnchor:SetPoint("TOPRIGHT", -27, -8)
+
+local importButton = CreateFrame("BUTTON", "WeakAurasTooltipImportButton", buttonAnchor, "UIPanelButtonTemplate")
+importButton:SetPoint("BOTTOMRIGHT", -8, 8)
+
+
+local showCodeButton = CreateFrame("BUTTON", "WeakAurasTooltipImportButton", buttonAnchor, "UIPanelButtonTemplate")
+showCodeButton:SetPoint("BOTTOMLEFT", 8, 8)
+showCodeButton:SetText(L["Show Code"])
+showCodeButton:SetWidth(90)
+
+local checkButtons, keyToButton, pendingData = {}, {}, {}
+for _, key in pairs(WeakAuras.internal_fields) do
+  keyToButton[key] = false
+end
+
+for index,category in pairs(WeakAuras.update_categories) do
+  local button = CreateFrame("checkButton", "WeakAurasTooltipCheckButton"..index, buttonAnchor, "ChatConfigCheckButtonTemplate")
+  for k, v in pairs(category) do
+    button[k] = v
   end
-  if(ItemRefTooltip.WeakAuras_Tooltip_Button) then
-    ItemRefTooltip.WeakAuras_Tooltip_Button:Hide();
+  _G[button:GetName().."Text"]:SetText(button.label)
+  button.func = function()
+    WeakAurasSaved.update_preferences[button.name] = button:GetChecked()
+    if button:GetChecked() ~= button.checkedStateForCopy then
+      pendingData.buttonsChecked = min(pendingData.buttonsChecked + 1, #checkButtons)
+    else
+      pendingData.buttonsChecked = max(pendingData.buttonsChecked - 1, 0)
+    end
+    WeakAuras.RefreshTooltipButtons()
   end
-  if(ItemRefTooltip.WeakAuras_Tooltip_Button2) then
-    ItemRefTooltip.WeakAuras_Tooltip_Button2:Hide();
+  checkButtons[index] = button
+  checkButtons[button.name] = button
+  for _, key in pairs(category.fields) do
+    keyToButton[key] = button
   end
-  if(ItemRefTooltip.WeakAuras_Desc_Box) then
-    ItemRefTooltip.WeakAuras_Desc_Box:Hide();
+end
+
+-- Ideally, we would not add an __index method to this
+-- But that would greatly bloat the update_category for display
+-- A better solution would be to refactor the data
+-- Such that all of the display tab settings are in their own subtable
+setmetatable(keyToButton,{__index = function(_, key) return key and checkButtons["display"] end,})
+
+local function install(data, oldData, patch, forceCopy)
```

Comment:
This functions should adjust hybridTables as needed, shouldn't it?

Reply:
Can you explain what you mean?